src/service/extensions-impl.js

```diff
@@ -268,6 +303,31 @@ export class Extensions {
   }
 
   /**
+   * Installs all shadow-root factories previously registered with
+   * `addShadowRootFactory_`.
+   * @param {!ShadowRoot} shadowRoot
+   * @param {!Array<string>} extensionIds
+   * @return {!Promise}
+   * @restricted
+   */
+  installFactoriesInShadowRoot(shadowRoot, extensionIds) {
```

Comment:
Does this handle the case of an extension that is not yet completed loading when the shadow root is installed?  Is there an upgrade path for that when it completes loading?